session/balance/provider/balance_tracker.go

```diff
@@ -66,12 +63,18 @@ func (bt *BalanceTracker) calculateBalance() {
 	bt.balance = bt.totalPromised - cost.Amount
 }
 
-// GetBalance returns the balance message
-func (bt *BalanceTracker) GetBalance() balance.Message {
+// GetBalance returns the current balance
+func (bt *BalanceTracker) GetBalance() uint64 {
 	bt.calculateBalance()
-	// TODO: sequence ID should come here, somehow
-	return balance.Message{
-		SequenceID: 0,
-		Balance:    bt.balance,
-	}
+	return bt.balance
+}
+
+// Start starts keeping track of time for balance
+func (bt *BalanceTracker) Start() {
+	bt.timeKeeper.StartTracking()
+}
+
+// Add increases the current balance by the given amount
+func (bt *BalanceTracker) Add(amount uint64) {
+	bt.totalPromised += amount
```

Comment:
Is there any possibility to have a race condition here for `totalPromised`? Maybe we need to protect it somehow?

Reply:
At the moment, I'm not at all concerned by race conditions here. But i'll add a lock to make it atomic anyway.